### PR TITLE
Region based nnunet scripts

### PR DESCRIPTION
**Pull request by plbenveniste**
In this PR we detail the scripts used for the training and evaluation of the region-based nnUNet. It contains:
- README.md : instructions to use the different files
- convert_BIDS_to_nnunet.py : script to convert data from the BIDS format to the nnUNet format
- convert_predictions_to_BIDS.py : script to convert the nnUNet predictions to the BIDS format using the conversion dictionnary
- evaluate_lesion_seg_prediction.py : this script computes the metrics using anima
- nnUNet_inference_analysis.ipynb : this notebook shows the performance of the nnunet with respect to its different folds
- convert_BIDS_to_nnunet_with_mul_PSIR.py : this last script was added to convert BIDS to nnunet format and multiplying PSIR images by -1

This list is to be updated.

### CODE REVIEW

File: nnunet/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+mkdir nnUNet_preprocessed
+mkdir nnUNet_results
+~~~
+
+We are training a region-based nnUNet taking an image from contrasts PSIR or STIR and creating a mask with 0=background, 1=spinal cord and 2=MS lesion.
+
+Convert the data to the nnUNet format :
+
+~~~
+python convert_bids_to_nnunet.py --path-data /path/to/BIDS/dataset --path-out /path/to/nnUNet_raw --taskname TASK-NAME --tasknumber DATASET-ID  --contrasts PSIR,STIR --test-ratio XX --time-point ses-XX --type training --exclude-file /path/to/exclude_file.yml
```
Comment (valosekj) [inline nnunet/README.md:60]:
```suggestion
python convert_BIDS_to_nnunet.py --path-data /path/to/BIDS/dataset --path-out /path/to/nnUNet_raw --taskname TASK-NAME --tasknumber DATASET-ID  --contrasts PSIR,STIR --test-ratio XX --time-point ses-XX --type training --exclude-file /path/to/exclude_file.yml
```

File: nnunet/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+echo "anima = ${HOME}/anima/Anima-Binaries-4.2/" >> .anima/config.txt
+echo "anima-scripts-public-root = ${HOME}/anima/Anima-Scripts-Public/" >> .anima/config.txt
+echo "extra-data-root = ${HOME}/anima/Anima-Scripts-Data-Public/" >> .anima/config.txt
+```
+
+## Installation of required libraries
+
+To install required libraries to train an nnUNet v2:
+
+```
```
Comment (valosekj) [inline nnunet/README.md:36]: just a nitpick, but comment/command that env has to be created before installing requirements could be added here

File: nnunet/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+~~~
+CUDA_VISIBLE_DEVICES=XXX nnUNetv2_train DATASET-ID CONFIG FOLD --npz
+~~~
+
+> **Note**
+> Example for Dataset 101, on 2d config on fold 0: CUDA_VISIBLE_DEVICES=2 nnUNetv2_train 101 2d 0 --npz
+
+# Model inference
+
+Convert data to nnUNet format for inference using `convert_bids_to_nnunet.py` or `convert_BIDS_to_nnunet_with_mul_PSIR.py` with `--type`=inference.
```
Comment (valosekj) [inline nnunet/README.md:96]:
```suggestion
Convert data to nnUNet format for inference using `convert_BIDS_to_nnunet.py` or `convert_BIDS_to_nnunet_with_mul_PSIR.py` with `--type`=inference`.
```

File: nnunet/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+We are training a region-based nnUNet taking an image from contrasts PSIR or STIR and creating a mask with 0=background, 1=spinal cord and 2=MS lesion.
+
+Convert the data to the nnUNet format :
+
+~~~
+python convert_bids_to_nnunet.py --path-data /path/to/BIDS/dataset --path-out /path/to/nnUNet_raw --taskname TASK-NAME --tasknumber DATASET-ID  --contrasts PSIR,STIR --test-ratio XX --time-point ses-XX --type training --exclude-file /path/to/exclude_file.yml
+~~~
+
+> **Note**
+> The test ratio is 0.2 for 20%. For M0 images, the time point is ses-M0.
```
Comment (valosekj) [inline nnunet/README.md:64]: > The test ratio is 0.2

Meaning that train ratio is 0.8, right?

File: nnunet/README.md (hunk 3 above)
Comment (valosekj) [reply nnunet/README.md:96]: How the datasets created using `type=training` vs `type=inference` differ? 
What is the purpose of `--test-ratio` for `type=inference`?

**Review by valosekj — APPROVED** [review]
LGTM! Thanks! I left only a few minor comments. Then the PR can be merged!

File: nnunet/README.md (hunk 2 above)
Comment (plbenveniste) [reply nnunet/README.md:46]: Done

**Pull request merged**